webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java

```diff
@@ -4160,7 +4160,38 @@ private String findSites(Map propertyCriteria) {
         return Xml.writeDocumentToString(dom);
     }
 
-
+    /**
+     * Get all site IDs for which the criteria fully or partially matches the title, the description or the skin.
+     *
+     * @param sessionid     valid session
+     * @param criteria      string to search for
+     * @return The site IDs
+     */
+    @WebMethod
+    @Path("/findSitesByTitle")
+    @Produces("text/plain")
+    @GET
+    public String findSitesByTitle(
+            @WebParam(name = "sessionid", partName = "sessionid") @QueryParam("sessionid") String sessionid,
+            @WebParam(name = "criteria", partName = "criteria") @QueryParam("criteria") String criteria) {
+        Session s = establishSession(sessionid);
+        String siteIDs = "";
+        try {
+            List<String> siteIdsList = siteService.getSiteIds(SelectionType.ANY, null, criteria,
+                    null, SortType.NONE, null);
+            StringBuilder sb = new StringBuilder();
+            if (siteIdsList != null && siteIdsList.size() > 0) {
+                for (String siteId : siteIdsList) {
+                    sb.append(siteId).append(",");
+                }
+                siteIDs = sb.substring(0, sb.length() - 1);
+            }
+        } catch (Throwable t) {
+            log.warn(this + ".findSiteIDs: Error encountered " + t.getMessage(), t);
```

Comment:
using old style logging please use the newer style:
```
log.warn("Error encountered {}", t.getMessage(), t);
```
this + ".findSiteIDs" is already added to the log message

Reply:
Alright. Using newer style now.